### PR TITLE
Support application/vnd.onshape.v2+json content-type

### PR DESCRIPTION
**Pull request by harmanpa**
Previously accept headers listed application/vnd.onshape.v1+json as well as json, responses are returning application/vnd.onshape.v2+json so need to include this in client

### CODE REVIEW

File: onshape-java/api-base/src/main/java/com/onshape/api/base/BaseClient.java [hunk 1]
```diff
@@ -331,7 +331,7 @@ public void setWorkingDir(File workingDir) {
             }
             throw new OnshapeException("No entity in response");
         }
-        if (response.getMediaType().toString().startsWith(MediaType.APPLICATION_JSON)) {
+        if (response.getMediaType().toString().startsWith(MediaType.APPLICATION_JSON) | response.getMediaType().toString().startsWith("application/vnd.onshape.v2+json")) {
```
Comment (awk) [inline onshape-java/api-base/src/main/java/com/onshape/api/base/BaseClient.java]: Please create a constant/final static for the "application/vnd.onshape.v2+json" string and use the constant through the file.

Comment (harmanpa) [reply onshape-java/api-base/src/main/java/com/onshape/api/base/BaseClient.java]: Good point, done. Should "v1" still be an option or is all json now "v2"?

**Comment by awk** [comment]
:shipit:

File: onshape-java/api-base/src/main/java/com/onshape/api/base/BaseClient.java (hunk 1 above)
Comment (awk) [reply onshape-java/api-base/src/main/java/com/onshape/api/base/BaseClient.java]: I wouldn't worry about v1 - the current stuff is using v2 and that's all the client needs to work with.

**Pull request merged**